src/Http/Controllers/WebhookController.php

```diff
@@ -42,26 +42,41 @@ public function store(Request $request)
         // Check if member email exists
         $member = User::findByEmail($email);
 
-        if (!$member) {
-            // Collect user data and subscribe to product
-            $user = [
-                'name' => $name,
-                'products' => $products,
-            ];
-            // Create user from customer email
-            $this->createUser($user, $email);
-        } else {
-            // Subscribe existing user to product
-            $memberProducts = $member->value('products') ?? [];
-            $memberProducts = collect($memberProducts)->merge($products)->unique()->sort()->values()->toArray();
-
-            $member->set('products', $memberProducts)
-            ->save();
+        if ($type == 'Order') {
+          if (!$member) {
+              // Collect user data and subscribe to product
+              $user = [
+                  'name' => $name,
+                  'products' => $products,
+              ];
+              // Create user from customer email
+              $this->createUser($user, $email);
+          } else {
+              // Subscribe existing user to product
+              $memberProducts = $member->value('products') ?? [];
+              $memberProducts = collect($memberProducts)->merge($products)->unique()->sort()->values()->toArray();
+
+              $member->set('products', $memberProducts)
+              ->save();
+          }
+
+          // Create Order model from request object
+          if (resolve(config('samcart.model'))::where('order_number', $slug)->count() == 0) {
+              $this->createOrder($request, $slug);
+          }
         }
 
-        // Create Order model from request object
-        if (resolve(config('samcart.model'))::where('order_number', $slug)->count() == 0) {
-            $this->createOrder($request, $slug);
+        // Remove products for refund, cancellations and delinquents
+        if ($type == 'Refund' || $type == 'SubscriptionDelinquent' || $type == 'Cancel' ) {
+          // Unsubscribe existing user from product
+          $memberProducts = $member->value('products') ?? [];
+
+          foreach($products as $product){
```

Comment:
I don't fully understand the context but I see potential that this is all products from a member and not just the one they cancelled or refunded or whatever... but I don't really understand the business process. You should test it :)

Reply:
Thanks! Tested and removes the cancelled product only. Also moved the comment for clarity.